File: src/main/java/eterea/core/api/rest/model/Moneda.java
/**
 * 
 */
package eterea.core.api.rest.model;

import java.io.Serializable;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@Entity
@Table(name = "moneda")
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
public class Moneda extends Auditable implements Serializable {
	/**
	* 
	*/
	private static final long serialVersionUID = 7521067423582119467L;

	@Id
	@Column(name = "mon_id")
	private Integer monedaId;

	@Column(name = "mon_nombre")
	@NotNull
	private String nombre = "";

	@Column(name = "mon_simbolo")
	@NotNull
	private String simbolo = "";

	@Column(name = "mon_cva_id")
	private Integer valorId;

}

File: src/main/java/eterea/core/api/rest/kotlin/model/Moneda.kt
package eterea.core.api.rest.kotlin.model

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table
data class Moneda(

    @Id @Column(name = "mon_id")
    var monedaId: Int? = null,

    @Column(name = "mon_nombre")
    var nombre: String? = "",

    @Column(name = "mon_simbolo")
    var simbolo: String? = "",

    @Column(name = "mon_cva_id")
    var valorId: Int? = null

) : Auditable()
